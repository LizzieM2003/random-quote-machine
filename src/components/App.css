body {
  margin: 0;
}

.App {
  font-family: sans-serif;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

#quote-box {
  background-color: #f5f5f5;
  width: 50%;
  height: 250px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  padding: 50px;
}

#quote {
  display: flex;
  justify-content: center;
  position: relative;
}

#text {
  font-size: 30px;
  font-weight: 500;
  text-align: center;
  margin: 0;
  width: 80%;
}

#author {
  width: 100%;
  text-align: right;
}

#quote-buttons {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

#new-quote {
  height: 40px;
  width: 120px;
  color: white;
  border-radius: 10px;
  font-size: 18px;
  outline: none;
}

@media (max-width: 850px) {
  #quote-box {
    width: 70%;
  }

  #text {
    font-size: 25px;
  }
}

@media (max-width: 450px) {
  #quote-box {
    width: 85%;
  }

  #text {
    font-size: 22px;
  }
}