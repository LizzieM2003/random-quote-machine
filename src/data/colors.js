export default [
  { colorOption: '#ff8533' },
  { colorOption: '#ff33ff' },
  { colorOption: '#ff6633' },
  { colorOption: '#3399ff' },
  { colorOption: '#669999' },
  { colorOption: '#b84dff' },
  { colorOption: '#c653c6' },
  { colorOption: '#538cc6' },
  { colorOption: '#99e600' },
  { colorOption: '#999966' },
  { colorOption: '#33cccc' },
  { colorOption: '#009999' },
  { colorOption: '#cc0000' },
  { colorOption: '#9933ff'},
  { colorOption: '#a300cc'}
];
