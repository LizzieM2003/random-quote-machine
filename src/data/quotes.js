export default [
  {
    quote: "It does not matter how slowly you go, as long as you don't stop.",
    author: 'Confucius'
  },
  {
    quote:
      "Life isn't about finding yourself. Life is about creating yourself.",
    author: 'George Bernard Shaw'
  },
  {
    quote:
      'Success is not final, failure is not fatal: it is the courage to continue that counts.',
    author: 'Winston Churchill'
  },
  {
    quote:
      'Education is the most powerful weapon which you can use to change the world.',
    author: 'Nelson Mandela'
  },
  {
    quote: 'Life is really simple, but we insist on making it complicated.',
    author: 'Confucius'
  },
  {
    quote: 'The secret of getting ahead is getting started.',
    author: 'Mark Twain'
  },
  {
    quote: 'If you can dream it, you can do it.',
    author: 'Walt Disney'
  },
  {
    quote: "Great things in business are never done by one person. They're done by a team of people.",
    author: 'Steve Jobs'
  },
  {
    quote: 'In the End, we remember not the words of our enemies, but the silence of our friends.',
    author: 'Martin Luther King Jr.'
  },
  {
    quote: 'Life is like riding a bicycle. To keep your balance, you must keep moving.',
    author: 'Albert Einstein'
  },
  {
    quote: 'Choose a job you love, and you will never have to work a day in your life.',
    author: 'Confucius'
  },
  {
    quote: "Believe you can and you're halfway there",
    author: 'Theodore Roosevelt'
  },
  {
    quote: 'Tears come from the heart and not from the brain.',
    author: 'Leonardo da Vinci'
  },
  {
    quote: 'No act of kindness, no matter how small, is ever wasted.',
    author: 'Aesop'
  },
  {
    quote: 'Success consists of going from failure to failure without loss of enthusiasm.',
    author: 'Winston Churchill'
  },
  {
    quote: 'Attitude is a little thing that makes a big difference',
    author: 'Winston Churchill'
  },
  {
    quote: 'You cannot escape the responsibility of tomorrow by evading it today',
    author: 'Abraham Lincoln'
  },
  {
    quote: 'When something is important enough, you do it even if the odds are not in your favour',
    author: 'Elon Musk'
  },
  {
    quote: 'The weak can never forgive. Forgiveness is the attribute of the strong.',
    author: 'Mahatma Gandhi'
  },
  {
    quote: 'It is far better to be alone, than to be in bad company.',
    author: 'George Washington'
  },
  {
    quote: 'Wise men speak because they have something to say; Fools because they have to say something.',
    author: 'Plato'
  },
  {
    quote: 'We know what we are, but know not what we may be.',
    author: 'William Shakespeare'
  },
  {
    quote: 'Kindness is the language which the deaf can hear and the blind can see.',
    author: 'Mark Twain'
  },
  {
    quote: 'We cannot solve our problems with the same thinking we used when we created them.',
    author: 'Albert Einstein'
  },
  {
    quote: 'A fool thinks himself to be wise, but a wise man knows himself to be a fool.',
    author: 'William Shakespeare'
  },
  {
    quote: 'No man has a good enough memory to be a successful liar.',
    author: 'Abraham Lincoln'
  },
  {
    quote: 'The time is always right to do what is right.',
    author: 'Martin Luther King Jr.'
  },
  {
    quote: "If you're going through hell, keep going.",
    author: 'Winston Churchill'
  },
  {
    quote: 'Leadership and learning are indispensible to each other',
    author: 'John F. Kennedy'
  },
  {
    quote: "The word 'happy' would lose its meaning if it were not balanced by sadness.",
    author: 'Carl Jung'
  },
  {
    quote: 'Our greatest glory is not in never falling, but in rising every time we fall.',
    author: 'Confucius'
  },
  {
    quote: 'The difference between stupidity and genius is that genius has its limits.',
    author: 'Albert Einstein'
  },
  {
    quote: 'Be sure you put your feet in the right place, then stand firm.',
    author: 'Abraham Lincoln'
  },
  {
    quote: "You have enemies? Good. That means you've stood up for something, sometime in your life.",
    author: 'Winston Churchill' 
  },
  {
    quote: 'Be kind whenever possible. It is always possible.',
    author: 'Dalai Lama'
  }
];
