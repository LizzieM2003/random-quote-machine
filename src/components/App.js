import React, { Component } from 'react';
import TwitterSVG from '../svgs/Twitter';
import QuotesSVG from '../svgs/Quotes';

import quotes from '../data/quotes';
import colors from '../data/colors';

import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      quote: quotes[0],
      color: colors[0]
    };

    this.handleNewQuote = this.handleNewQuote.bind(this);
  }

  handleNewQuote() {
    const newQuoteIndex = Math.floor(Math.random() * quotes.length);
    const newQuote = quotes[newQuoteIndex];
    const newColorIndex = Math.floor(Math.random() * colors.length);
    const newColor = colors[newColorIndex];

    this.setState({ quote: newQuote, color: newColor });
  }

  render() {
    const tweetText = encodeURIComponent(this.state.quote.quote);
    const appStyles = {
      backgroundColor: this.state.color.colorOption,
      color: this.state.color.colorOption
    };

    return (
      <div className="App" style={appStyles}>
        <div id="quote-box">
          <div id="quote">
            <QuotesSVG id="quote-svg" color={this.state.color.colorOption} />
            <p id="text">{this.state.quote.quote}</p>
          </div>
          <p id="author">{this.state.quote.author}</p>
          <div id="quote-buttons">
            <a
              id="tweet-quote"
              href={'https://twitter.com/intent/tweet?text=' + tweetText}
              target="_blank"
            >
              <TwitterSVG color={this.state.color.colorOption} />
            </a>
            <button
              id="new-quote"
              onClick={this.handleNewQuote}
              style={{
                backgroundColor: this.state.color.colorOption,
                border: `1px solid ${this.state.color.colorOption}`
              }}
            >
              New Quote
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
