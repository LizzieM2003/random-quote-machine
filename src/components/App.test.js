import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './App';
import quotes from '../data/quotes';
import colors from '../data/colors';

configure({ adapter: new Adapter() });

describe('App', () => {
  const app = shallow(<App />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('contains a wrapping element of id `quote-box`', () => {
    expect(app.find('#quote-box').exists()).toBe(true);
  });

  it('contains an element with id `text`', () => {
    expect(app.find('#text').exists()).toBe(true);
  });

  it('contains an element with id `author` inside `quote-box`', () => {
    expect(app.find('#quote-box').children('#author').length).toBe(1);
  });

//   it('contains an element with id `tweet-quote` inside `quote-box`', () => {
//     expect(app.find('#quote-box').children('#tweet-quote').length).toBe(1);
//   });

  it('initialises the `state` with a quote', () => {
    expect(app.state().quote).toEqual(quotes[0]);
  });

  it('initialises the `state` with a color scheme', () => {
    expect(app.state().color).toEqual(colors[0]);
  });

  describe('when clicking the `New Quote` button', () => {
    beforeEach(() => {
      app.find('#new-quote').simulate('click');
    });

    it('updates the `state` with a new quote', () => {
      expect(app.state().quote).not.toEqual(quotes[0]);
    });

    it('changes the color scheme', () => {
      expect(app.state().color).not.toEqual(colors[0]);
    });
  });
});
