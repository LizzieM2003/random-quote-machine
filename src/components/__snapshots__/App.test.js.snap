// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="quote-box"
>
        <div
                id="quote"
        >
                <quotes
                        color="#ff8533"
                        id="quote-svg"
                />
                <p
                        id="text"
                >
                        It does not matter how slowly you go, as long as you don't stop.
                </p>
        </div>
        <p
                id="author"
        >
                Confucius
        </p>
        <div
                id="quote-buttons"
        >
                <a
                        href="https://twitter.com/intent/tweet?text=It%20does%20not%20matter%20how%20slowly%20you%20go%2C%20as%20long%20as%20you%20don't%20stop."
                        id="tweet-quote"
                        target="_blank"
                >
                        <twitter
                                color="#ff8533"
                        />
                </a>
                <button
                        id="new-quote"
                        onClick={[Function]}
                        style={
                                Object {
                                        "backgroundColor": "#ff8533",
                                        "border": "1px solid #ff8533",
                                      }
                        }
                >
                        New Quote
                </button>
        </div>
</div>,
      "className": "App",
      "style": Object {
        "backgroundColor": "#ff8533",
        "color": "#ff8533",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="quote"
>
            <quotes
                        color="#ff8533"
                        id="quote-svg"
            />
            <p
                        id="text"
            >
                        It does not matter how slowly you go, as long as you don't stop.
            </p>
</div>,
          <p
            id="author"
>
            Confucius
</p>,
          <div
            id="quote-buttons"
>
            <a
                        href="https://twitter.com/intent/tweet?text=It%20does%20not%20matter%20how%20slowly%20you%20go%2C%20as%20long%20as%20you%20don't%20stop."
                        id="tweet-quote"
                        target="_blank"
            >
                        <twitter
                                    color="#ff8533"
                        />
            </a>
            <button
                        id="new-quote"
                        onClick={[Function]}
                        style={
                                    Object {
                                                "backgroundColor": "#ff8533",
                                                "border": "1px solid #ff8533",
                                              }
                        }
            >
                        New Quote
            </button>
</div>,
        ],
        "id": "quote-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <quotes
                color="#ff8533"
                id="quote-svg"
/>,
              <p
                id="text"
>
                It does not matter how slowly you go, as long as you don't stop.
</p>,
            ],
            "id": "quote",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "#ff8533",
                "id": "quote-svg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "It does not matter how slowly you go, as long as you don't stop.",
                "id": "text",
              },
              "ref": null,
              "rendered": "It does not matter how slowly you go, as long as you don't stop.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Confucius",
            "id": "author",
          },
          "ref": null,
          "rendered": "Confucius",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="https://twitter.com/intent/tweet?text=It%20does%20not%20matter%20how%20slowly%20you%20go%2C%20as%20long%20as%20you%20don't%20stop."
                id="tweet-quote"
                target="_blank"
>
                <twitter
                                color="#ff8533"
                />
</a>,
              <button
                id="new-quote"
                onClick={[Function]}
                style={
                                Object {
                                                "backgroundColor": "#ff8533",
                                                "border": "1px solid #ff8533",
                                              }
                }
>
                New Quote
</button>,
            ],
            "id": "quote-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <twitter
                  color="#ff8533"
/>,
                "href": "https://twitter.com/intent/tweet?text=It%20does%20not%20matter%20how%20slowly%20you%20go%2C%20as%20long%20as%20you%20don't%20stop.",
                "id": "tweet-quote",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "#ff8533",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New Quote",
                "id": "new-quote",
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "#ff8533",
                  "border": "1px solid #ff8533",
                },
              },
              "ref": null,
              "rendered": "New Quote",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="quote-box"
>
          <div
                    id="quote"
          >
                    <quotes
                              color="#ff8533"
                              id="quote-svg"
                    />
                    <p
                              id="text"
                    >
                              It does not matter how slowly you go, as long as you don't stop.
                    </p>
          </div>
          <p
                    id="author"
          >
                    Confucius
          </p>
          <div
                    id="quote-buttons"
          >
                    <a
                              href="https://twitter.com/intent/tweet?text=It%20does%20not%20matter%20how%20slowly%20you%20go%2C%20as%20long%20as%20you%20don't%20stop."
                              id="tweet-quote"
                              target="_blank"
                    >
                              <twitter
                                        color="#ff8533"
                              />
                    </a>
                    <button
                              id="new-quote"
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "backgroundColor": "#ff8533",
                                                  "border": "1px solid #ff8533",
                                                }
                              }
                    >
                              New Quote
                    </button>
          </div>
</div>,
        "className": "App",
        "style": Object {
          "backgroundColor": "#ff8533",
          "color": "#ff8533",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="quote"
>
              <quotes
                            color="#ff8533"
                            id="quote-svg"
              />
              <p
                            id="text"
              >
                            It does not matter how slowly you go, as long as you don't stop.
              </p>
</div>,
            <p
              id="author"
>
              Confucius
</p>,
            <div
              id="quote-buttons"
>
              <a
                            href="https://twitter.com/intent/tweet?text=It%20does%20not%20matter%20how%20slowly%20you%20go%2C%20as%20long%20as%20you%20don't%20stop."
                            id="tweet-quote"
                            target="_blank"
              >
                            <twitter
                                          color="#ff8533"
                            />
              </a>
              <button
                            id="new-quote"
                            onClick={[Function]}
                            style={
                                          Object {
                                                        "backgroundColor": "#ff8533",
                                                        "border": "1px solid #ff8533",
                                                      }
                            }
              >
                            New Quote
              </button>
</div>,
          ],
          "id": "quote-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <quotes
                  color="#ff8533"
                  id="quote-svg"
/>,
                <p
                  id="text"
>
                  It does not matter how slowly you go, as long as you don't stop.
</p>,
              ],
              "id": "quote",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "#ff8533",
                  "id": "quote-svg",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "It does not matter how slowly you go, as long as you don't stop.",
                  "id": "text",
                },
                "ref": null,
                "rendered": "It does not matter how slowly you go, as long as you don't stop.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Confucius",
              "id": "author",
            },
            "ref": null,
            "rendered": "Confucius",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="https://twitter.com/intent/tweet?text=It%20does%20not%20matter%20how%20slowly%20you%20go%2C%20as%20long%20as%20you%20don't%20stop."
                  id="tweet-quote"
                  target="_blank"
>
                  <twitter
                                    color="#ff8533"
                  />
</a>,
                <button
                  id="new-quote"
                  onClick={[Function]}
                  style={
                                    Object {
                                                      "backgroundColor": "#ff8533",
                                                      "border": "1px solid #ff8533",
                                                    }
                  }
>
                  New Quote
</button>,
              ],
              "id": "quote-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <twitter
                    color="#ff8533"
/>,
                  "href": "https://twitter.com/intent/tweet?text=It%20does%20not%20matter%20how%20slowly%20you%20go%2C%20as%20long%20as%20you%20don't%20stop.",
                  "id": "tweet-quote",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "color": "#ff8533",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New Quote",
                  "id": "new-quote",
                  "onClick": [Function],
                  "style": Object {
                    "backgroundColor": "#ff8533",
                    "border": "1px solid #ff8533",
                  },
                },
                "ref": null,
                "rendered": "New Quote",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
